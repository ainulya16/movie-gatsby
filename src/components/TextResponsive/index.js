import React from 'react';
import styled from 'styled-components';
import { bool, node } from 'prop-types';

const DefaultWrapper = styled.div`
  margin: 5px 0px;
`;

const ResponsiveTitleText = styled.h1`
  font-size: 42pt;

  /* Extra small devices (phones, less than 768px) */
  @media (max-width: 768px) {
    font-size: 18pt;
  }
  /* Small devices (tablets, 768px and up) */
  @media (min-width: 768px) and (max-width: 992px) {
    font-size: 22pt;
  }
  /* Medium devices (desktops, 992px and up) */
  @media (min-width: 992px) and (max-width: 1200px) {
    font-size: 32pt;
  }
  /* Large devices (large desktops, 1200px and up) */
  @media (min-width: 1200px) {
    font-size: 42pt;
  }
`;

const ResponsiveContentText = styled.span`
  font-size: 12pt;

  /* Extra small devices (phones, less than 768px) */
  @media (max-width: 768px) {
    font-size: 12pt;
  }
  /* Small devices (tablets, 768px and up) */
  @media (min-width: 768px) and (max-width: 992px) {
    font-size: 12pt;
  }
  /* Medium devices (desktops, 992px and up) */
  @media (min-width: 992px) and (max-width: 1200px) {
    font-size: 14pt;
  }
  /* Large devices (large desktops, 1200px and up) */
  @media (min-width: 1200px) {
    font-size: 14pt;
  }
`;

/**
 * Title components.
 * Use for every title.
 *
 * @param { Object } props
 */
const TextResponsive = ({ children, isTitle }) => {
  // eslint-disable-next-line max-len
  const TextComponent = isTitle ? (
    <ResponsiveTitleText>{children}</ResponsiveTitleText>
  ) : (
    <ResponsiveContentText>{children}</ResponsiveContentText>
  );

  return <DefaultWrapper>{TextComponent}</DefaultWrapper>;
};

TextResponsive.propTypes = {
  children: node.isRequired,
  isTitle: bool,
};

TextResponsive.defaultProps = {
  isTitle: false,
};

export default TextResponsive;
