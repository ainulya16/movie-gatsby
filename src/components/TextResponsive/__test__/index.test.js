/* eslint-env jest */
import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import TextResponsive from '../index';

describe('Text Responsive', () => {
  it('should renders TextResponsive ', () => {
    const wrapper = shallow(<TextResponsive>Some Children</TextResponsive>);
    expect(wrapper.exists()).toBe(true);
  });
  it('should renders TextResponsive title', () => {
    const wrapper = shallow(
      <TextResponsive isTitle>Some Children</TextResponsive>,
    );
    expect(wrapper.exists()).toBe(true);
  });
  it('should renders TextResponsive content', () => {
    const wrapper = shallow(<TextResponsive>Some Children</TextResponsive>);
    expect(wrapper.exists()).toBe(true);
  });
  it('should match the snapshot', () => {
    const tree = renderer
      .create(<TextResponsive>Some Children</TextResponsive>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
