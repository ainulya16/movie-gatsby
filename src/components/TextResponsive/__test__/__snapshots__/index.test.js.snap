// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Responsive should match the snapshot 1`] = `
.c0 {
  margin: 5px 0px;
}

.c1 {
  font-size: 12pt;
}

@media (max-width:768px) {
  .c1 {
    font-size: 12pt;
  }
}

@media (min-width:768px) and (max-width:992px) {
  .c1 {
    font-size: 12pt;
  }
}

@media (min-width:992px) and (max-width:1200px) {
  .c1 {
    font-size: 14pt;
  }
}

@media (min-width:1200px) {
  .c1 {
    font-size: 14pt;
  }
}

<div
  className="c0"
>
  <span
    className="c1"
  >
    Some Children
  </span>
</div>
`;
